### ISSUE TITLE
[Collateral] Create full test suite for all functions in Collateral.ts

### ISSUE THREAD

**Issue by pelsasser** [MARKETProtocol/MARKET.js#53]
## Before you `start work`

Please read our contribution [guidelines](https://docs.marketprotocol.io/#contributing) and if there is a bounty involved  please also see [here](https://docs.marketprotocol.io/#gitcoin-and-bounties)

If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please make sure to comment in the issue here immediately after starting work so we know your plans for implementation and a timeline.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

### User Story
As a dev I would like testing to know if I break something!

### Why Is this Needed?

*Summary*: Increased coverage across MARKET.js

### Description

*Type*: Feature

### Current Behavior
Limited testing for Collateral.ts functionality

### Expected Behavior
Near 100% coverage!

### Reproduction
`npm run test` after setting up test environment and running migrations (or use docker!)

### Solution

*Summary*: Create tests for Collateral.ts

### Definition of Done
Create Tests For
- depositCollateralAsync (note a call to ERC20.approve is needed, see other tests for examples)
  - call is successful
  - balance after deposit is correct
- getUserAccountBalanceAsync
- withdrawCollateralAsync
  - call is successful
  - balance after is correct
- settleAndCloseAsync
  - call fails
  - attempting to get this to work in testing is a bit beyond the scope here so can be ifnored for now.

## Bounty Hunters - please let us know your planned solution and time frame so we can approve.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.3 ETH (131.08 USD @ $436.94/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/MARKET.js/53/708).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $29,652.65 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by subramanianv** [comment]
Hi @pelsasser I would like to work on this bounty. I am going to model it based on existing tests. I am hoping to finish it by wednesday next week or hopefully before that :)

**Comment by eswarasai** [comment]
@subramanianv -- Please apply on Gitcoin bounty before starting work so that we can approve and there's no confusion in the future. Thanks!

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 6 days, 9 hours from now. Please review their questions below:


1. [subramanianv](https://gitcoin.co/profile/subramanianv) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/MARKETProtocol/MARKET.js/53/708?mutate_worker_action=approve&worker=subramanianv) | [reject worker](https://gitcoin.co/issue/MARKETProtocol/MARKET.js/53/708?mutate_worker_action=reject&worker=subramanianv))_. 	
 * Q: Posted in the comments

**Comment by eswarasai** [comment]
@subramanianv -- Thanks for applying. Feel free to start work on the task. I'll let @pelsasser approve you once he's online. Please do join our [Discord](https://marketprotocol.io/discord) channel to ask any queries that you might have.

**Comment by subramanianv** [comment]
@pelsasser @eswarasai Please review the PR

**Comment by subramanianv** [comment]
@eswarasai I am also not able to sign the CLA. The CLA app keeps redirecting itself to the agreement page again and again.

**Comment by pelsasser** [comment]
@subramanianv - hmm, okay I will look into the CLA issue, thanks for reporting back on it!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 0.3 ETH (137.0 USD @ $456.67/ETH) has been submitted by__:

 1. [@subramanianv](https://gitcoin.co/profile/subramanianv)
 2. [@subramanianv](https://gitcoin.co/profile/subramanianv)



@pelsasser please take a look at the submitted work:
* [PR](https://github.com/MARKETProtocol/MARKET.js/pull/58) by @subramanianv
* [PR](https://github.com/MARKETProtocol/MARKET.js/pull/58) by @subramanianv

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/MARKET.js/53/708)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
$16,659.80 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by pelsasser** [comment]
@subramanianv - thanks for submitting work, I just paid this out, may take a bit to get processed!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 0.3 ETH (143.0 USD @ $476.67/ETH) attached to this issue has been approved & issued to @subramanianv.__  

 * Learn more at [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/MARKET.js/53/708)
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $17,135.46 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
